dlrover/python/training_event/exporter.py

```diff
@@ -0,0 +1,315 @@
+# Copyright 2025 The DLRover Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import atexit
+import logging
+import os
+import sys
+import threading
+import time
+from abc import abstractmethod
+from queue import Empty, Full, Queue
+from typing import Optional
+
+from dlrover.python.training_event.config import (
+    get_default_config,
+    get_default_logger,
+)
+from dlrover.python.training_event.event import Event
+
+_CONFIG = get_default_config()
+_LOGGER = get_default_logger()
+
+
+class EventExporter:
+    """
+    EventExporter is the interface to export the event to the target.
+
+    By default, the exporter is singleton in a process, the exporter
+    implementation should ensure the thread safety.
+    """
+
+    @abstractmethod
+    def export(self, event: Event):
+        pass
+
+    @abstractmethod
+    def close(self):
```

Comment:
If an abstraction for "close" is defined, it is recommended to also have an abstraction for "start".